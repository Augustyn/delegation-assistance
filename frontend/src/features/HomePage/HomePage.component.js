import React from "react";

import LayoutMain from "../../components/layouts/LayoutMain/LayoutMain.container";

const HomePage = props => {
  return (
    <LayoutMain title="Home page">
      <section>
        <div>Home page</div>
      </section>
    </LayoutMain>
  );
};

export default HomePage;
