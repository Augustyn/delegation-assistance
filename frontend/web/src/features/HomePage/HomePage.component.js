import React from "react";

import LayoutMain from "../../components/layouts/LayoutMain/LayoutMain.component";

const HomePage = props => {
  return (
    <LayoutMain>
      <div>Home page</div>
      <a href="/login">Login to app</a>
    </LayoutMain>
  );
};

export default HomePage;
