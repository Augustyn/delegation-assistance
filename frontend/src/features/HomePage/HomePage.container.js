import React, { Component } from "react";
import HomePage from "./HomePage.component";

class HomePageContainer extends Component {
  render() {
    return (
      <div headerClass="index">
        <HomePage />
      </div>
    );
  }
}

export default HomePageContainer;
