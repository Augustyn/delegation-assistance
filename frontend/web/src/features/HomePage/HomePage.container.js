import React, { Component } from "react";
import { connect } from "react-redux";

import HomePage from "./HomePage.component";

class HomePageContainer extends Component {
  render() {
    return <HomePage />;
  }
}

export default connect(
  null,
  null
)(HomePageContainer);
