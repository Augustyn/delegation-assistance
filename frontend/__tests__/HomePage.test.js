import React from "react";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import HomePage from "../src/features/HomePage/HomePage.component";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const props = {
    handleSubmit: jest.fn(),
    subbmiting: false,
    invalid: false
  };

  const enzymeWrapper = shallow(<HomePage {...props} />);

  return {
    props,
    enzymeWrapper
  };
};

describe("Home Page", () => {
  let enzymeWrapper;
  let props;

  beforeEach(() => {
    ({ enzymeWrapper, props } = setup());
  });

  it("should render correctly", () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
});
