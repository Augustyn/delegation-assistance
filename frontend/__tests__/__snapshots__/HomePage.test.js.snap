// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render correctly 1`] = `
<Connect(LayoutMainContainer)
  title="Home page"
>
  <section>
    <div>
      Home page
    </div>
  </section>
</Connect(LayoutMainContainer)>
`;
